Project Management Process.md

```diff
@@ -0,0 +1,287 @@
+# IPFS Project Management Process
+
+**Status: work in progress, not even a full draft yet**
+
+## Introduction
+TODO: describe the purpose of this document
+
+- this is an opt-in process, nobody in the community is *required* to use this process
+
+## Table on Contents
+- [TL;DR](#tldr)
+- [Projects](#projects)
+- [Roles](#roles)
+  - [Product Owners](#product-owners)
+  - [Project Leads](#project-leads)
+  - [Project Managers](#project-managers)
+- [Structure](#structure)
+  - [Overview](#overview)
+  - [Goals](#goals)
+  - [Backlog](#backlog)
+  - [Milestones](#milestones)
+  - [Project Roadmap](#project-roadmap)
+  - [Organization Roadmap](#organization-roadmap)
+- [Workflow](#flow)
+  - [Pipelines](#pipelines)
+    - [Stages](#stages)
+    - [Flow](#flow)
+    - [Example](#example)
+  - [Weekly Updates](#weekly-updates)
+  - [Roadmap Updates](#roadmap-updates)
+- [Implementation Guideline](#implementation-guideline)
+
+## TL;DR
+
+The IPFS project management process has the following **cadence**:
+
+Weekly Updates
+- once a week (2 weeks?)
+- discuss last week's progress
+- discuss the plans for next week
+
+Roadmap Updates
+- every 3 months
+- identify, discuss and plan the next steps for the organization as a whole
+
+The IPFS project management process defines the following **units of work**:
+
+Projects 
+- may contain other projects
+- have a project leader who "roadmaps" (decides what are goals and milestones, and when they should happen)
+- have a roadmap (list of milestones and goals)
+- have a backlog (list of goals)
+
+Milestones 
+- measurable unit of progress
+- contains a list of goals, ordered by priority
+- usually but not always sequentially ordered
+- may have a deadline
+
+Goals
+- an individual actionable unit of work
+- may have other Goals as dependencies
+- may have parts or sub-goals
+- may have parts or sub-goals big enough to merit own Goals
+
+The IPFS project management process, planning and tracking work is done with the following **tools**:
+
+Roadmaps
+- are attached to a project
+- have a list of milestones, ordered by completion time
+
+Backlogs
+- are attached to a project
+- have a list of all goals for a project, ordered by priority
+
+Pipelines
+- have a set of stages through which goals move (the steps between "not done" and "done")
+- show what has been completed, what's being worked on at the moment and what will be worked on next
+
+
+## Projects
+TODO
+
+## Roles
+TODO
+
+#### Project Leads
+TODO
+
+#### Product Owners
+TODO
+
+#### Project Managers
+TODO
```

Comment:
Having thought about this, I would actually leave out the Product Owners part (not used anywhere else in the document and can be different per project, and as an organization, who's a product owner and who's not?). 

Perhaps leave out the project managers part, too as this would be more an individual roles on the participating organizations. Or should we try to describe the roles of project managers?